compiler/luci/service/src/Nodes/CircleFullyConnected.cpp

```diff
@@ -36,4 +40,95 @@ luci::CircleNode *CloneNodeLet<CN::DEF>::visit(const luci::CircleFullyConnected
   return cloned;
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleFullyConnected *node)
+{
+  auto input_shape = circle_shape(loco::must_cast<CircleNode *>(node->input()));
+  auto weights_shape = circle_shape(loco::must_cast<CircleNode *>(node->weights()));
+
+  loco::TensorShape out_shape;
+
+  // NOTE Some recipes in some repositories are using rank 4 input for FullyConnected.
+  //      Until they are all fixed, disable following assert.
+  // TODO Enable following assert after related fixes are applied
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L194
+  // LUCI_ASSERT(input_shape.rank() == 2 || input_shape.rank() == 3,
+  //             "Input rank of FullyConnected should be 2 or 3");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L225
+  LUCI_ASSERT(weights_shape.rank() == 2, "Weights of FullyConnected should be 2");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L353-L367
+
+  /*
+   * [CircleNode]       [CircleNode]
+   *   (input)            (weight)
+   *         \            /
+   *     [CircleFullyConnected]
+   *               |
+   *            (output)
+   *
+   * **Pre-conditions:**
+   *    input_shape.rank() <= 4
+   *      * remark: TFLite allows <=3 ranks, but there are rank 4 input recipes in ONE
+   *    weight_shape.rank() == 2 and all dimensions are known.
+   *    Regardless of whether input_shape[-1] is known or not, input_shape[-1] == weight_shape[-1]
```

Comment:
This sentence could make some confusion. 

It could be interpreted in two ways : 
  1. If `input_shape[-1]` is unknown, `weight_shape[-1]` should be also unknown.
      If `input_shape[-1]` and `weight_shape[-1]` are known, they should be the same value. 

  2. When runtime `input_shape[-1]` and `weight_shape[-1]` are both known, it should be same value

Reply:
Oh, I see. I think line77 has to be changed.
How about replacing line 77 with the second option you mentioned?
```
When runtime (input_shape[-1] and weight_shape[-1] are both known), it should be same value
```